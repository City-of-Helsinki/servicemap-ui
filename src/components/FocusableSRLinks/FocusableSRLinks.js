import React from 'react';
import PropTypes from 'prop-types';

const FocusableSRLinks = ({ classes, items }) => (
  <div className={classes.srFocusedContainer}>
    {
      items.map(v => (
        <a href={v.href} className={classes.srFocused} key={v.href}>
          {v.text}
        </a>
      ))
    }
  </div>
);

FocusableSRLinks.propTypes = {
  classes: PropTypes.shape({
    srFocusedContainer: PropTypes.string,
    srFocused: PropTypes.string,
  }).isRequired,
  items: PropTypes.arrayOf(PropTypes.shape({
    href: PropTypes.string.isRequired,
    text: PropTypes.node.isRequired,
  })).isRequired,
};

export default FocusableSRLinks;
