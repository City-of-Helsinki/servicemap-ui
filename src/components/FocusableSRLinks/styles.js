export default theme => ({
  srFocusedContainer: {
    position: 'absolute',
    alignItems: 'center',
    backgroundColor: '#fff',
    borderRadius: '0 2px 2px 0',
    boxShadow: '0 2px 2px 0 rgba(0,0,0,.3), 0 0 0 1px rgba(0,0,0,.2)',
    display: 'flex',
    margin: '110px auto 8px 0',
    overflow: 'hidden',
    zIndex: theme.zIndex.infront,
  },
  srFocused: {
    border: `5px solid ${theme.palette.primary.main}`,
    borderRadius: '2px',
    color: '#2228af',
    cursor: 'pointer',
    display: 'inline-block',
    fontSize: '14px',
    lineHeight: '20px',
    margin: '8px 12px',
    minHeight: '32px',
    textDecoration: 'underline',
    textAlign: 'center',
    width: '106px',
    outline: 'none',
    '&:not(:focus)': {
      clip: 'rect(1px,1px,1px,1px)',
      overflow: 'hidden',
      position: 'absolute',
      padding: 0,
    },
  },
});
