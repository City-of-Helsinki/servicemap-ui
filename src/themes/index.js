import { createMuiTheme } from '@material-ui/core';

const focusIndicator = {
  outline: '2px solid transparent',
  boxShadow: '0 0 0 2px rgb(255, 255, 255), 0 0 0 6px rgb(71, 131, 235), 0 0 0 8px rgb(255, 255, 255)',
};

const focusIndicatorDark = {
  outline: '2px solid transparent',
  boxShadow: '0 0 0 2px rgb(255, 255, 255), 0 0 0 6px rgb(0, 0, 0), 0 0 0 8px rgb(255, 255, 255)',
};

const props = {
  // Globally disable all default mui focus effects
  MuiButtonBase: {
    disableRipple: true,
  },
  MuiButton: {
    disableRipple: true,
    disableFocusRipple: true,
  },
  MuiTypography: {
    variant: 'body2',
  },
};

const overrides = theme => ({
  MuiButtonBase: {
    root: {
      // Default keyboard focus indicator for buttons
      '&$focusVisible': theme === 'dark' ? focusIndicatorDark : focusIndicator,
    },
  },
  MuiInputBase: {
    // Default keyboard focus indicator for input fields
    root: { '&$focused': theme === 'dark' ? focusIndicatorDark : focusIndicator },
  },
  MuiListItemText: {
    root: {
      marginTop: 0,
      marginBottom: 0,
    },
  },
  MuiSvgIcon: {
    root: {
      fontSize: 24,
    },
  },
  PrivateSwitchBase: {
    root: {
      padding: 12,
    },
  },
});

const breakpoints = {
  values: {
    xs: 0,
    sm: 700,
    md: 900,
    lg: 1280,
    xl: 1920,
  },
};

const spacing = 8;

const custom = {
  body2light: {
    color: 'rgba(0,0,0,0.7)',
  },
};

const zIndex = {
  behind: -1,
  forward: 50,
  sticky: 51,
  infront: 900,
};

const typography = {
  useNextVariants: true,
  fontSize: 16,
  // Use the system font instead of the default Roboto font.
  fontFamily: [
    'Lato',
  ].join(','),
  body1: {
    fontSize: '1.043rem',
    letterSpacing: '0.03125rem', // 0.5px
    lineHeight: '1.75rem',
  },
  body2: {
    fontSize: '0.913rem',
    letterSpacing: '0.015625rem', // 0.25px
    lineHeight: '1.25rem',
  },
  subtitle1: {
    fontSize: '1.03rem',
    fontWeight: 'bold',
    letterSpacing: '0.00937rem', // 0.15px
    lineHeight: '1.5rem',
  },
  caption: {
    fontSize: '0.7725rem', // 12.36px
    fontWeight: 'bold',
    letterSpacing: '0.025rem', // 0.4px
    lineHeight: '1rem',
    color: 'rgba(0,0,0,0.6)',
  },
  h1: {
    fontSize: '6.259rem',
    letterSpacing: '-0.09375em', // -1.5px
  },
  h2: {
    fontSize: '3.862rem',
    fontWeight: 'bold',
    letterSpacing: '-0.03125rem', // -0.5px
  },
  h3: {
    fontSize: '3.089rem',
    fontWeight: 'bold',
    letterSpacing: 0,
  },
  h4: {
    fontSize: '2.214rem',
    letterSpacing: '0.015625rem', // 0.25px
  },
  h5: {
    fontSize: '1.564rem',
    letterSpacing: 0,
  },
  h6: {
    fontSize: '1.288rem',
    fontWeight: 'bold',
    letterSpacing: '0.015625rem', // 0.25px
    lineHeight: '1.5rem',
  },
  button: {
    fontSize: '0.90125rem', // 14.42px
    fontWeight: 'bold',
    letterSpacing: '0.078125rem', // 1.25px
    lineHeight: '1rem', // 16px
    color: '#ffffff',
  },
  overline: {
    fontSize: '0.7725rem', // 12.36px
    fontWeight: 'bold',
    letterSpacing: '0.125rem', // 2px
    lineHeight: '1rem',
  },
};

// Color palette for normal theme
export const paletteDefault = {
  primary: {
    main: 'rgb(25, 100, 230)',
    highContrast: '#fff',
  },
  secondary: {
    main: 'rgb(43, 47, 57)',
    hover: '#1d39ad',
  },
  background: {
    main: 'linear-gradient(340.58deg, rgb(11, 123, 237) 0%, rgb(20, 108, 232) 67.04%, rgb(25, 100, 230) 100%)',
    plain: 'rgb(20, 108, 232)',
    front: 'linear-gradient(340.58deg, rgb(11, 123, 237) 0%, rgb(20, 108, 232) 67.04%, rgb(25, 100, 230) 100%)',
  },
  white: {
    light: '#f2f2f2',
    main: '#FFFFFF',
    dark: '#949494',
    contrastText: '#000',
  },
  detail: {
    main: 'rgb(25, 100, 230)',
    alpha: 'rgba(25,100,230,0.5)',
  },
  disabled: {
    main: 'rgb(239, 239, 239)',
    strong: '#898989',
  },
  focusBorder: {
    main: 'rgb(71, 131, 235)',
  },
  warning: {
    main: '#AF0021',
  },
};

// Color palette for dark theme
export const paletteDark = {
  primary: {
    main: '#353638',
    highContrast: '#fff',
  },
  secondary: {
    main: 'rgb(43, 47, 57)',
    hover: '#5b5d61',
  },
  background: {
    main: '#4A4A4C',
    plain: '#4A4A4C',
    front: 'linear-gradient(326.21deg, rgba(0, 0, 0, 0.79) 0%, rgba(71, 71, 71, 0.79) 100%)',
  },
  white: {
    light: '#f2f2f2',
    main: '#FFFFFF',
    dark: '#949494',
    contrastText: '#000',
  },
  detail: {
    main: '#585B63',
    alpha: 'rgba(88,91,99,0.5)',
  },
  disabled: {
    main: 'rgb(239, 239, 239)',
    strong: '#898989',
  },
  warning: {
    main: '#AF0021',
  },
  focusBorder: {
    main: '#000',
  },
};

// Themes
const SMTheme = createMuiTheme({
  props,
  overrides: overrides('default'),
  breakpoints,
  typography,
  spacing,
  custom,
  palette: paletteDefault,
  zIndex,
});

const SMThemeDark = createMuiTheme({
  props,
  overrides: overrides('dark'),
  breakpoints,
  typography,
  spacing,
  custom,
  palette: paletteDark,
  zIndex,
});


export default { SMTheme, SMThemeDark };
