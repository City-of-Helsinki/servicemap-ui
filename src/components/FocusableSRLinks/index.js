import { withStyles } from '@material-ui/core';
import styles from './styles';
import FocusableSRLinks from './FocusableSRLinks';

export default withStyles(styles)(FocusableSRLinks);
