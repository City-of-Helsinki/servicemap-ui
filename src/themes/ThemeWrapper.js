import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { ThemeProvider } from '@mui/material/styles';
import themes from '.';

const mapStateToProps = (state) => {
  const { user } = state;
  return {
    theme: user.theme,
  };
};

// Component to handle theme changes
const ThemeHandler = ({ children, theme }) => (
// Get correct theme setting from store
  <ThemeProvider theme={theme === 'dark' ? themes.SMThemeDark : themes.SMTheme}>
    {children}
  </ThemeProvider>
);
ThemeHandler.propTypes = {
  children: PropTypes.node,
  theme: PropTypes.string.isRequired,
};
ThemeHandler.defaultProps = {
  children: null,
};

const ThemeWrapper = connect(
  mapStateToProps,
)(ThemeHandler);

export default ThemeWrapper;
