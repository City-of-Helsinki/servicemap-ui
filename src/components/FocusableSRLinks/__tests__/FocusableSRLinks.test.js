// Link.react.test.js
import React from 'react';
import { createMount } from '@material-ui/core/test-utils';
import { MuiThemeProvider } from '@material-ui/core';
import { IntlProvider, FormattedMessage } from 'react-intl';
import themes from '../../../themes';
import FocusableSRLinks from '../index';

// Generic required props for SimpleListItem
const mockProps = {
  items: [
    {
      href: '#test-href',
      text: 'Title text',
      // text: <FormattedMessage id="fm.test" />,
    },
  ],
};

// Mock props for intl provider
const intlMock = {
  locale: 'en',
  messages: {
    'fm.test': 'FM test text',
  },
};

// eslint-disable-next-line react/prop-types
const Providers = ({ children }) => (
  <IntlProvider {...intlMock}>
    <MuiThemeProvider theme={themes.SMTheme}>
      {children}
    </MuiThemeProvider>
  </IntlProvider>
);

describe('<FocusableSRLinks />', () => {
  // let render;
  // let shallow;
  let mount;

  beforeEach(() => {
    // shallow = createShallow({ wrappingComponent: Providers });
    mount = createMount({ wrappingComponent: Providers });
  });

  it('should work', () => {
    const component = mount(<FocusableSRLinks {...mockProps} />);
    expect(component).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const component = mount(<FocusableSRLinks {...mockProps} />);

    let link = component.find('a');
    expect(link.text()).toEqual(mockProps.items[0].text);

    // Check text rendering with FormattedMessage
    const component2 = mount(
      <FocusableSRLinks
        items={[
          {
            href: '#test-href',
            text: <FormattedMessage id="fm.test" />,
          },
        ]}
      />,
    );
    // Dive through FormattedMessage and consumers
    link = component2.find(FormattedMessage);
    expect(link.text()).toEqual(intlMock.messages['fm.test']);
  });

  it('does set href correctly', () => {
    const component = mount(<FocusableSRLinks {...mockProps} />);

    const link = component.find('a');
    expect(link.props().href).toEqual(mockProps.items[0].href);
  });
});
