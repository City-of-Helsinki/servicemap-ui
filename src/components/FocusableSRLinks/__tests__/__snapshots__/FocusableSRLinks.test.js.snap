// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FocusableSRLinks /> should work 1`] = `
<WithStyles(FocusableSRLinks)
  items={
    Array [
      Object {
        "href": "#test-href",
        "text": "Title text",
      },
    ]
  }
>
  <FocusableSRLinks
    classes={
      Object {
        "srFocused": "FocusableSRLinks-srFocused-2",
        "srFocusedContainer": "FocusableSRLinks-srFocusedContainer-1",
      }
    }
    items={
      Array [
        Object {
          "href": "#test-href",
          "text": "Title text",
        },
      ]
    }
  >
    <div
      className="FocusableSRLinks-srFocusedContainer-1"
    >
      <a
        className="FocusableSRLinks-srFocused-2"
        href="#test-href"
        key="#test-href"
      >
        Title text
      </a>
    </div>
  </FocusableSRLinks>
</WithStyles(FocusableSRLinks)>
`;
